catalog/core/catalog-core-dynamic-api/src/main/java/ddf/catalog/data/dynamic/api/MetacardPropertyDescriptor.java

```diff
@@ -0,0 +1,110 @@
+    /**
+     * Returns whether this property should be indexed by the underlying catalog source.
+     * @return whether this property should be indexed by the underlying catalog source
+     */
+    boolean isIndexedBySource();
+
+    /**
+     * Set the flag for this attribute indicating if this attribute should be indexed by the
+     * underlying catalog source.
+     * @param indexedBySource true if this attribute should be indexed
+     */
+    void setIndexedBySource(boolean indexedBySource);
+
+    /**
+     * Indicates that this is a tokenized representation of the data. If
+     * true, the attribute should be tokenized before storing.
+     * @return true if the attribute data is received in a tokenized format
+     */
+    boolean isTokenized();
+
+    /**
+     * Set the flag indicating that this attribute is a tokenized representation of the data. If
+     * true, the attribute should be tokenized before storing.
+     * @param tokenized true if this attribute data is tokenized upon receipt
+     */
+    void setTokenized(boolean tokenized);
+
+    /**
+     * Indicates that the catalog source should storing the value of this property.
+     * @return true if the catalog source should be storing the value of this property
+     */
+    boolean isStored();
+
+    /**
+     * Sets the flag indicating that the catalog source should store the value of this property.
+     */
+    void setStored(boolean stored);
+
+    /**
+     * Returns the attribute format for this property. Especially useful for properties with the
+     * same underlying class type (e.g., STRING, GEOMETRY, XML).
+     * @return the {@link ddf.catalog.data.AttributeType.AttributeFormat} for this property
+     */
+    AttributeType.AttributeFormat getFormat();
+
+    /**
+     * Sets the attribute format for this property. This is useful when the property has a common
+     * representation class but is semantically different from other types, e.g. STRING, GEOMETRY,
+     * XML).
+     * @param format the format of this property
+     */
+    void setFormat(AttributeType.AttributeFormat format);
+
+    // Add the methods from the DynaProperty class
+
+    /**
+     * Gets the (optional) type of the indexed content for DynaProperty's that support this feature.
+     * @return the Class for the content type if this is an indexed DynaProperty and this feature is supported. Otherwise null.
+     */
+    Class<?> getContentType();
+
+    /**
+     * Get the name of this property.
+     * @return the name of the property
+     */
+    String getName();
+
+    /**
+     * Gets the Java class representing the data type of the underlying property values.
+     * @return the property type
+     */
+    Class<?> getType();
+
+    /**
+     * Does this property represent an indexed value (ie an array or List)?
+     * @return true if the property is indexed (i.e. is a List or array), otherwise false
+     */
+    boolean isIndexed();
```

Comment:
Agreed that it is a bit confusing, but I just used the same name as the underlying DynaProperty class to ease the implementation. The API is not tied to the implementation, but having the method named the same removes a lot of proxying code. I'm open to renaming it, but would need to be convinced.